src/playlist-loader.js

```diff
@@ -340,7 +340,8 @@ export const updateMaster = (master, newMedia, unchangedCheck = isPlaylistUnchan
  *         The time in ms to wait before refreshing the live playlist
  */
 export const refreshDelay = (media, update) => {
-  const lastSegment = media.segments[media.segments.length - 1];
+  const segments = media.segments || [];
```

Comment:
Fixes an issue I was seeing in our multi period source where sometimes media does not have segments. I can make this a different pull request, but it was easy enough.